lib/juttle-job.js

```diff
@@ -0,0 +1,170 @@
+var _ = require('underscore');
+var Base = require('extendable-base');
+var JSDP = require('juttle-jsdp');
+var JSDPValueConverter = require('./jsdp-value-converter');
+var child_process = require('child_process');
+var Promise = require('bluebird');
+var JuttledErrors = require('./errors');
+var logger = require('log4js').getLogger('juttle-job');
+var events = require('backbone').Events;
+
+// This class handles the details of running a program from a
+// bundle, handling an explicit stop from a job manager, emitting the
+// proper events when the job has completed, etc.
+//
+// Actually passing the results of programs back to websocket and/or
+// http clients is handled by subclasses. Based on the method, the Job
+// Manager will create a WebsocketJuttleJob or HTTPJuttleJob object.
```

Comment:
`s/HTTPJuttleJob/ImmediateJuttleJob/g`? Though not sure if it makes sense to refer to subclasses in a superclass.